plugins/kotlin/idea/tests/test/org/jetbrains/kotlin/idea/actions/KotlinAddImportActionTest.kt

```diff
@@ -0,0 +1,85 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.idea.actions
+
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.descriptors.SimpleFunctionDescriptor
+import org.jetbrains.kotlin.idea.base.test.TestRoot
+import org.jetbrains.kotlin.idea.test.InTextDirectivesUtils
+import org.jetbrains.kotlin.idea.test.KotlinLightCodeInsightFixtureTestCase
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+import org.jetbrains.kotlin.test.TestMetadata
+import org.junit.internal.runners.JUnit38ClassRunner
+import org.junit.runner.RunWith
+import java.io.File
+
+@TestRoot("idea/tests")
+@RunWith(JUnit38ClassRunner::class)
+@TestMetadata("testData/idea/actions/kotlinAddImportAction")
+class KotlinAddImportActionTest : KotlinLightCodeInsightFixtureTestCase() {
```

Comment:
Usually we create smth like `AbstractKotlinAddImportActionTest` that has public `doTest` function and add it to [Fe10GenerateTests](https://github.com/JetBrains/intellij-community/blob/a79368860d6d3c7d558d861d40df319eca910b6e/plugins/kotlin/util/test-generator-fe10/test/org/jetbrains/kotlin/fe10/testGenerator/Fe10GenerateTests.kt#L793) 

it this case you can't forget to add/miss newly added test data cases.

you can regenerate tests with `Generate Kotlin Tests (K1 + K2)` run configurations.

Reply:
Thanks for the explanation! I had seen that that class existed, but didn't understand how it was used. This helps.